File: perfect_power.cpp
#include <iostream>
#include <cmath>

using namespace std;

struct P {
	int base;
	int power;
};

P isPP(int n) {
	for (int m = 2; m <= int(ceil(sqrt(n))); ++m) {
		cout << m << endl;
		for (int k = 2;k <= int(ceil(log2(n))); ++k) {
			cout << k << " ";
			if (pow(m, k) == n) {
				return {m, k};
			}
		}
		cout << endl;		
	}
	return {0, 0};
}

int main() {
	auto fPP = isPP(81);
	int powerlist [2] = {fPP.base, fPP.power};

	cout << "\n\n[" << powerlist[0] << ", " << powerlist[1] << "]" << endl;
}

File: perfect power/perfect_power.py
from math import sqrt, log2, floor, ceil

def isPP(n):
	for m in range(2, ceil(sqrt(n))+1):
		print(m)
		for k in range(2, ceil(log2(n))+1):
			if m**k == n:
				print("{} {}".format(m, k))
				return (m, k)



x = isPP(81)

print(x)